aws_lambda_powertools/utilities/data_classes/event_bridge_event.py

```diff
@@ -67,3 +68,22 @@ def detail(self) -> Dict[str, Any]:
     def replay_name(self) -> Optional[str]:
         """Identifies whether the event is being replayed and what is the name of the replay."""
         return self["replay-name"]
+
+    @property
+    def event_size(self) -> int:
+        """Returns the EventBridge Event Size.
+
+        URLs:
+            https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-putevent-size.html
+        """
+
+        size = 0
```

Comment:
Consider making use of caching.

Due to string conversion being a relatively expensive operation, check if event size was previously calculated before running through.

Reply:
I guess that being said, does it make more sense to move it to its own class?

Reply:
Could you share how do you plan on using it? I suspect this would be best utilized as a function that receives an EventBridge event, because this dataclass is "immutable" (no setters). 

The usefulness of checking the event size is to prevent sending an event that will be rejected by EventBridge. This data class would highly likely contain an event that is already under 256K, unless you're using it differently hence the question ;)